### DISCUSSION TITLE
[Idea]: Auto export folder

### DISCUSSION THREAD

**Discussion by maskiranbedbug** [sircharlo/meeting-media-manager#3470] [Ideas] [answered]
### What is your question?

Why was the feature of downloading media to a folder removed? Am I missing something in the new version that I can't seem to locate such folder, or that option is removed altogether? 

### Is your question related to a problem?

No

### Additional context

I am in a congregation that records the meeting for stream on our language. For recording we are using the vMix software which is provided by the branch, and we are required to import all the meeting files in the vMix window before each meeting. The previous version had a handy option to automatically download files in a directory, and the way our setup worked, after boot the program would start, auto download media and auto quit, so by the time any human interaction is needed, the media was all prepared.
This feature really made the work easy, even for those who are not so proficient with computer.

Now we're really struggling and I am hoping to get a way around this so that we can continue to use this software.
I tried to install the previous version and prevent major update but with no success. 
I also tried looking in the source code in effort to find the auto update url and block it via windows host file, also with no success.

Please provide feedback on this, or an option how to handle our unique situation.
Thank you very much in advance.

### Confirmations

- [X] I confirm that I have read the documentation website.
- [X] I confirm that I have searched through the existing discussions to make sure my question is not a duplicate.

**Comment by sircharlo** [comment]
We could definitely add an option to automatically export meeting media.
I would tuck this away in the advanced options, as this is not typical use by any means, but it's definitely possible.

**Reply by maskiranbedbug** [reply]
I would honestly prefer to have it as a setting, that way it would be possible for the additional congregation that shares the hall to use the same software, and to actually use it for what it provides, not only downloading media :)
Thank you for even taking this into consideration.

**Comment by mtdvlpr** [comment]
You should just be able to download [v24.4.8](https://github.com/sircharlo/meeting-media-manager/releases/tag/v24.4.8) and disable auto updates in the settings while waiting for this feature to become available in the new version.

**Reply by mtdvlpr** [reply]
You should enable the `Disable non-critical automatic app updates` setting and the `Disable checking for updates` setting.

**Reply by maskiranbedbug** [reply]
That was my first though exactly. 
I did download the last version before the upgrade and disabled the non-critical updates and disabled checking for updates, however those settings didn't seem to apply because on the initial app startup, the program checked for updates and automatically downloaded the last version, saying it will be installed on program close.

I did try to disable internet access while setting the app - to disable the updates in settings - but once internet was enabled, it checked for the latest version and downloaded it in the background.
Any suggestion to get around this?

**Reply by mtdvlpr** [reply]
It might download it in the background, but it should not actually install it. Does it really update after closing or is it only saying that it will?

**Reply by maskiranbedbug** [reply]
Initially it only said that the update is downloaded and will not be automatically installed (because of setting that prevents auto installing updates), but while I was trying multiple times if it will auto update or not by closing and opening... it ended up installing in the end without my consent...
And I was only trying that to simulate multiple meetings going by, because I can't have someone be at the desk, and the app auto updates - and even worse, that I am not in the hall at that time.

Is it possible for me to create a build out of a certain version where i will have removed the code that does the auto updates, at least until this gets sorted some other way in the current release?

**Reply by mtdvlpr** [reply, marked as answer]
I've updated the release artifacts for [v24.4.8](https://github.com/sircharlo/meeting-media-manager/releases/tag/v24.4.8) (Windows only) with a fix for this. Please try if this build fixed it for you.

**Reply by mtdvlpr** [reply]
Double check that the settings are correctly saved. You might have to enable `Disable checking for updates` before enabling `Disable non-critical automatic app updates` in order for the settings to be correctly saved.

**Reply by maskiranbedbug** [reply]
@mtdvlpr Whatever you did with that update did the trick. At least on my personal laptop. Now that I have "Disable checking for updates" and "Disable non-critical automatic app updates", when I open the program, it doesn't automatically start downloading the new version and I don't get any toast message about installing after closing.
I will give it a try at the hall's PC this weekend, but I'm quite sure that we won't have any issue going forward.
Thank you very much for the swift resolution on this, it will bring great joy when the rest of the team finds out that the issue is solved and we get to keep our workflow as simple as possible.

**Reply by mtdvlpr** [reply]
Glad to hear it!  The old version does miss some essential fetching logic and isn't future ready, so stay tuned for a new update that features an export folder again. We'll keep you posted here.

**Comment by maskiranbedbug** [comment]
Amazing! Thank you for already starting your work on this.

Now that I'm thinking about this, the new version of the program with this enhancement will greatly solve us another issue, namely when brothers share their JW Library playlist, we go to the trouble of unzipping the file and then unzipping the content in order to find images attached to their talk.
Now the way I see it, we will be able to import playlist into M3, then export all of that to external folder.
I hope that with this we get two birds with one stone :)

Thank you very much

**Reply by mtdvlpr** [reply]
Yes, that's right! Although, jwl playlists are also supported in the old version. If you add the playlist file to the correct date through M3, it will unpack it and place the files in the correct order in the output folder.

**Reply by mtdvlpr** [reply]
That's better than unzipping yourself, because some files (like song videos) are not inside the zip, but rather referenced in the sqlite database. M3 fetches those as well for you.

**Reply by sircharlo** [reply]
> Amazing! Thank you for already starting your work on this.
Pleasure! It will be released in the next version of the app!


> when brothers share their JW Library playlist, we go to the trouble of unzipping the file and then unzipping the content in order to find images attached to their talk

Why are you doing it this way? The app can import JWPLAYLIST files with a simple drag and drop, no manual unzipping required. Have you tried it?

**Comment by ux-git** [comment]
@maskiranbedbug does vMix fully replace JW Library for you? Does it present media on the secondary displays as well?

**Reply by maskiranbedbug** [reply]
The way our setup works is that we only use vMix, because we need full control over the 2nd screen, mainly because of zoom.
So we have a picture background that makes up the yeartext, and that is by default displayed on the 2nd screen. We record and display camera on vMix, route it externally to become "input" for camera and audio to zoom, and only present full screen pictures and videos to the 2nd screen when needed.
Additionally, we have an input into vMix that presents the content of 2nd screen, so when we have zoom talk, we present the zoom window on 2nd screen for audience to see, and record the TV input into vMix for recording.

**Reply by ux-git** [reply]
Thanks for sharing!

**Comment by sircharlo** [comment]
The initial implementation of the "export to folder" feature will be released in the next update. 
I'm looking forward to your feedback!

**Reply by maskiranbedbug** [reply]
Amazing news! I am looking forward to seeing this new update.
I will be sure to provide immediate feedback :)

Thank you again for your effort, it is very appreciated.
